### PR TITLE
Add more payment info. Decline on large amounts. Parse accept/decline.

### PR DESCRIPTION
**Pull request by philwinder**

### CODE REVIEW

File: front-end/js/client.js [hunk 1]
```diff
@@ -43,28 +43,37 @@ function resetTags() {
...
     $.ajax({
         url: "orders",
         type: "POST",
         async: false,
         success: function (data, textStatus, jqXHR) {
-            console.log("Order placed.");
-            alert("Order placed!");
-            deleteCart();
+            console.log('Headers: ' + JSON.stringify(jqXHR));
+            console.log('Headers: ' + jqXHR);
```
Comment (idcrosby) [inline front-end/js/client.js]: Probably want to remove these

File: orders/src/main/java/works/weave/socks/orders/controllers/OrdersController.java [hunk 2]
```diff
@@ -114,12 +144,18 @@ CustomerOrder newOrder(@RequestBody NewOrderResource item) {
...
 //    }
 
-    private String parseId(String href) {
-        Pattern idPattern = Pattern.compile("[\\w-]+$");
-        Matcher matcher = idPattern.matcher(href);
-        if (!matcher.find()) {
-            throw new IllegalStateException("Could not parse user ID from: " + href);
+    private float calculateTotal(List<Item> items) {
+        float amount = 0F;
+        float shipping = 4.99F;
```
Comment (idcrosby) [inline orders/src/main/java/works/weave/socks/orders/controllers/OrdersController.java:149]: The shipping amount is hard coded in both the front end and back end at the moment(?) we might trip over this later

File: payment/service.go [hunk 3]
```diff
@@ -11,15 +13,31 @@ type Authorisation struct {
...
+	if amount <= s.declineOverAmount {
+		authorised = true
+	}
+	return Authorisation{
+		Authorised: authorised,
+	}, nil
 }
+
+var ErrZeroPayment = errors.New("Zero payment")
+var ErrNegativePayment = errors.New("Negative payment")
```
Comment (idcrosby) [inline payment/service.go]: Might be better to merge these two cases

File: payment/transport.go [hunk 4]
```diff
@@ -40,11 +43,47 @@ func encodeError(_ context.Context, err error, w http.ResponseWriter) {
 }
 
 func decodeAuthoriseRequest(_ context.Context, r *http.Request) (interface{}, error) {
-	return authoriseRequest{}, nil
+	// Read the content
+	var bodyBytes []byte
+	if r.Body != nil {
+		bodyBytes, _ = ioutil.ReadAll(r.Body)
```
Comment (idcrosby) [inline payment/transport.go]: should check/handle the error here

**Comment by idcrosby** [comment]
LGTM, just curious, do you have a specific use case in mind for adding the max payment limit or is it just to make the service more interesting?

**Comment by philwinder** [comment]
Thanks, will fix. Yes, I just wanted to make it more interesting. I wanted to add a step which could legitimately fail and stop an order. I was going to make it randomly decline payments, but I thought that was too complicated. I thought that > amount was simpler but exercised the idea of an unauthorised payment.

File: orders/src/main/java/works/weave/socks/orders/controllers/OrdersController.java (hunk 2 above)
Comment (philwinder) [inline orders/src/main/java/works/weave/socks/orders/controllers/OrdersController.java:149]: Yeah. Todo. Not really a big issue right now though. See #119

**Pull request merged**